cmd/storj-sdk/network.go

```diff
@@ -0,0 +1,179 @@
+	processes := NewProcesses()
+
+	const (
+		host            = "127.0.0.1"
+		gatewayPort     = 9000
+		satellitePort   = 10000
+		storageNodePort = 11000
+	)
+
+	defaultSatellite := net.JoinHostPort(host, strconv.Itoa(satellitePort+0))
+
+	arguments := func(name, command, addr string, rest ...string) []string {
+		return append([]string{
+			"--log.level", "debug",
+			"--config-dir", ".",
+			command,
+			"--identity.server.address", addr,
+		}, rest...)
+	}
+
+	for i := 0; i < satelliteCount; i++ {
+		name := fmt.Sprintf("satellite/%d", i)
+
+		dir := filepath.Join(dir, "satellite", fmt.Sprint(i))
+		if err := os.MkdirAll(dir, 0644); err != nil {
+			return nil, err
+		}
+
+		process, err := processes.New(name, "satellite", dir)
+		if err != nil {
+			return nil, utils.CombineErrors(err, processes.Close())
+		}
+		process.Info.Address = net.JoinHostPort(host, strconv.Itoa(satellitePort+i))
+
+		process.Arguments["setup"] = arguments(name, "setup", process.Info.Address)
+		process.Arguments["run"] = arguments(name, "run", process.Info.Address,
+			"--kademlia.bootstrap-addr", defaultSatellite,
+		)
+	}
+
+	gatewayArguments := func(name, command string, addr string, rest ...string) []string {
+		return append([]string{
+			"--log.level", "debug",
+			"--config-dir", ".",
+			command,
+			"--identity.server.address", addr,
+		}, rest...)
+	}
+
+	for i := 0; i < satelliteCount; i++ {
+		name := fmt.Sprintf("gateway/%d", i)
+
+		dir := filepath.Join(dir, "gateway", fmt.Sprint(i))
+		if err := os.MkdirAll(dir, 0644); err != nil {
+			return nil, err
+		}
+
+		satellite := processes.List[i]
+
+		process, err := processes.New(name, "gateway", dir)
+		if err != nil {
+			return nil, utils.CombineErrors(err, processes.Close())
+		}
+		process.Info.Address = net.JoinHostPort(host, strconv.Itoa(gatewayPort+i))
+
+		process.Arguments["setup"] = gatewayArguments(name, "setup", process.Info.Address,
+			"--satellite-addr", satellite.Info.Address,
+		)
+		process.Arguments["run"] = gatewayArguments(name, "run", process.Info.Address)
+	}
+
+	for i := 0; i < storageNodeCount; i++ {
+		name := fmt.Sprintf("storage/%d", i)
+
+		dir := filepath.Join(dir, "storage", fmt.Sprint(i))
+		if err := os.MkdirAll(dir, 0644); err != nil {
+			return nil, err
+		}
+
+		process, err := processes.New(name, "storagenode", dir)
+		if err != nil {
+			return nil, utils.CombineErrors(err, processes.Close())
+		}
+		process.Info.Address = net.JoinHostPort(host, strconv.Itoa(storageNodePort+i))
+
+		process.Arguments["setup"] = arguments(name, "setup", process.Info.Address,
+			"--piecestore.agreementsender.overlay-addr", defaultSatellite,
+		)
+		process.Arguments["run"] = arguments(name, "run", process.Info.Address,
+			"--piecestore.agreementsender.overlay-addr", defaultSatellite,
+			"--kademlia.bootstrap-addr", defaultSatellite,
+			"--kademlia.farmer.email", fmt.Sprintf("storage%d@example.com", i),
+			"--kademlia.farmer.wallet", "0x0123456789012345678901234567890123456789",
```

Comment:
these changed from farmer to operator.

bummer this is kind of brittle, but i don't know what else to do. we should make sure running these codepaths are part of builds

Reply:
we have some things still to do before we can start using it, but once we have it running we can switch captplanet integration test for storj-sdk test.